pytest_catchlog.py

```diff
@@ -122,6 +134,15 @@ def __init__(self, config):
         self.formatter = logging.Formatter(
                 get_option_ini(config, 'log_format'),
                 get_option_ini(config, 'log_date_format'))
+        terminal = py.io.TerminalWriter(sys.stderr)  # pylint: disable=no-member
+        self.console = logging.StreamHandler(terminal)
+        self.console.setFormatter(self.formatter)
+        # Add the handler to logging
+        logging.root.addHandler(self.console)
```

Comment:
If someone attaches a handler, there must be a point where it is eventually detached. The logging subsystem can't track loading/unloading of plugins by itself unfortunately, but it is necessary to provide reentrancy (at least to be able to run `pytester` in-process).

I would suggest the same approach as used in abusalimov/pytest-catchlog@4121487982a1b39937f9daa04cfc94c016e201bc:
```diff
@@ -123,13 +123,27 @@ def __init__(self, config):
+        handler = logging.StreamHandler()
+        handler.setFormatter(self.formatter)
+
+        self.handler = handler
+
+    @pytest.mark.hookwrapper
+    def pytest_runtestloop(self, session):
+        """Runs all collected test items."""
+        with catching_logs(self.handler):
+            yield  # run all the tests
+
```